src/components/hooks/useNamedEntity.ts

```diff
@@ -0,0 +1,76 @@
+import { useAPIContext } from 'components/data/apiContext';
+import { Core } from 'flyteidl';
+import {
+    DomainIdentifierScope,
+    getNamedEntity,
+    NamedEntity,
+    RequestConfig,
+    ResourceType
+} from 'models';
+import { useFetchableData } from './useFetchableData';
+import { usePagination } from './usePagination';
+
+export interface UseNamedEntityInput {
+    resourceType: Core.ResourceType;
+    project: string;
+    domain: string;
+    name: string;
+}
+
+/** Fetches a NamedEntity (Workflow, LaunchPlan, Task, etc) for a given
+ * resourceType/project/domain/name. This is useful to determine any metadata
+ * associated with the given entity name */
+export function useNamedEntity(input: UseNamedEntityInput) {
+    return useFetchableData<NamedEntity, UseNamedEntityInput>(
+        {
+            debugName: 'NamedEntity',
+            defaultValue: {} as NamedEntity,
+            doFetch: id => getNamedEntity(id),
+            useCache: true
+        },
+        input
+    );
+}
+
+/** Fetches the NamedEntity record for a LaunchPlan */
+export function useLaunchPlanNamedEntity(
```

Comment:
Are these extra hooks intended to be in this PR? I don't see them used (yet)

Reply:
I'm being proactive. They will be used in an upcoming feature.